File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/fight/GameFightUpdateTeamMessage.scala





















// Generated on 05/03/2013 10:34:40
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameFightUpdateTeamMessage extends MessageDeserializer {
  val messageId = 5572

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GameFightUpdateTeamMessage.messageId

    val fightId: Short
    val team: FightTeamInformations.TypeImpl

    def serialize(buf: Buffer) {
      buf.writeShort(fightId)
      team.serialize(buf)
    }

  }

  protected def create(fightId: Short, team: FightTeamInformations.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val fightId = buf.readShort()
    val team = FightTeamInformations.deserialize(buf)

    create(fightId, team)
  }

}

object GameFightUpdateTeamMessage extends GameFightUpdateTeamMessage {
  case class MessageImpl(fightId: Short, team: FightTeamInformations.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(fightId: Short, team: FightTeamInformations.TypeImpl) = MessageImpl(fightId, team)

  def apply(fightId: Short, team: FightTeamInformations.TypeImpl) = create(fightId, team)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameFightUpdateTeamMessage.java


// Generated on 05/08/2013 19:37:44
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameFightUpdateTeamMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5572;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short fightId;
    public FightTeamInformations team;
    
    public GameFightUpdateTeamMessage() { }
    
    public GameFightUpdateTeamMessage(short fightId, FightTeamInformations team) {
        this.fightId = fightId;
        this.team = team;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(fightId);
        team.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        fightId = buf.readShort();
        if (fightId < 0)
            throw new RuntimeException("Forbidden value on fightId = " + fightId + ", it doesn't respect the following condition : fightId < 0");
        team = new FightTeamInformations();
        team.deserialize(buf);
    }
    
}
